design/synchronizer/auto-failover.md

```diff
@@ -0,0 +1,92 @@
+# Synchronizer Auto-Failover
+
+The following is an overview of the workflow for a Vault Sychronizer to gain leadership:
+
+![](./activity-diagram.png)
+
+Given the following Policy:
+```yml
+- !policy
+  id: conjur
+  body:
+  - !policy
+    id: vault-synchronizer
+    body:
+    - !variable leader-expiration
+
+    # The Group all Synchronizer hosts need to be added to.
```

Comment:
How can I do that?
I see a few use cases here

1. First installation
- Synchronizer installer creates this policy
- Synchronizer installer creates another policy for safes, lobs etc altogether with Synchronizer host
- Synchronizer installer needs to update this policy - with host how?

2. Installation of second instance of Synchronizer
- Synchronizer installer updates another policy for safes, lobs etc altogether with new Synchronizer host
- Synchronizer installer needs to update this policy - with host how?

3. A few Synchronizer installers already exists - manual update needed
- This policy is created manually 
- How to add already existing hosts here

4. Synchronizer uninstall
- How to remove host from this group

Reply:
@GrzegorzK-CA, I put together a working example in the Conjur Intro project: https://github.com/conjurdemos/conjur-intro/blob/vault-synchronizer/bin/dap#L428.  

The only thing missing here is setting the JSON document with safe names and last updated dates. Bash does not like the double quotes of JSON. Because the JSON is passed through so many functions, I was not able to get it escaped correctly. I updated the variable via the UI to make sure it worked as expected.  You should have no problem doing this with the .NET SDK.